models/perm/access/repo_permission.go

```diff
@@ -102,45 +102,28 @@ func (p *Permission) CanWriteIssuesOrPulls(isPull bool) bool {
 	return p.CanWrite(unit.TypeIssues)
 }
 
-// ColorFormat writes a colored string for these Permissions
-func (p *Permission) ColorFormat(s fmt.State) {
-	noColor := log.ColorBytes(log.Reset)
-
-	format := "perm_model.AccessMode: %-v, %d Units, %d UnitsMode(s): [ "
-	args := []interface{}{
-		p.AccessMode,
-		log.NewColoredValueBytes(len(p.Units), &noColor),
-		log.NewColoredValueBytes(len(p.UnitsMode), &noColor),
-	}
-	if s.Flag('+') {
-		for i, unit := range p.Units {
-			config := ""
-			if unit.Config != nil {
-				configBytes, err := unit.Config.ToDB()
-				config = string(configBytes)
-				if err != nil {
-					config = err.Error()
-				}
+func (p *Permission) LogString() string {
+	format := "<Permission AccessMode=%s, %d Units, %d UnitsMode(s): [ "
+	args := []any{p.AccessMode.String(), len(p.Units), len(p.UnitsMode)}
```

Comment:
```suggestion
	args := make([]any, 0, 3 + 5*len(p.Units) + 2*len(p.UnitsMode))
	args = append(args, p.AccessMode.String(), len(p.Units), len(p.UnitsMode)}
```

Reply:
According to many tests, the "pre-allocation" doesn't improve noticeable performance.

And here it doesn't affect performance, I prefer to keep readability instead of the unclear math trick.